haystack/utils/hf.py

```diff
@@ -241,7 +241,7 @@ def check_valid_model(model_id: str, model_type: HFModelType, token: Optional[Se
         allowed_model = model_info.pipeline_tag in ["sentence-similarity", "feature-extraction"]
         error_msg = f"Model {model_id} is not a embedding model. Please provide a embedding model."
     elif model_type == HFModelType.GENERATION:
-        allowed_model = model_info.pipeline_tag in ["text-generation", "text2text-generation"]
+        allowed_model = model_info.pipeline_tag in ["text-generation", "text2text-generation", "image-text-to-text"]
```

Comment:
This seems like a pretty small change. Should we add this in a separate PR since it's probably good to have regardless of the new component?

Reply:
I needed this to use Llama Guard 4, which is multimodal (we don't currently use this feature).
LMK if I should open another PR for this change.